### ISSUE TITLE
PHPUnit Unit Testing?

### ISSUE THREAD

**Issue by holisticnetworking** [DevinVinson/WordPress-Plugin-Boilerplate#500]
I see lots of conversation about this from 2014, but then it seems to have gone dark thereafter.

Any chance now is a good time to revisit this question?

**Comment by rajanit2000** [comment]
@holisticnetworking You are right. Will start to implement the unit testing module. And release it next major version.

**Comment by dingo-d** [comment]
I wouldn't go above defining the base `phpunit.xml.dist` file for this boilerplate. The code in it is super tightly coupled. You'd have to mock tons of stuff to make it work, trust me, not worth the work.

**source**: I followed this boilerplate and had to run unit and integration tests for a client, and I wanted to shoot myself and am now refactoring everything with interfaces and better structure in general 😆
